compiler-core/src/parse.rs

```diff
@@ -1163,8 +1163,21 @@ where
                 } else {
                     None
                 };
-                let (end, rsqb_e) =
-                    self.expect_one_following_series(&Token::RightSquare, "a pattern")?;
+
+                let (end, rsqb_e) = match self.maybe_one(&Token::RightSquare) {
+                    Some((start, end)) => Ok((start, end)),
+                    None => {
+                        if tail.is_some() {
+                            self.next_tok_error(ParseErrorType::ListPatternAfterSpread)
+                        } else {
```

Comment:
Just doing this would produce confusing errors when another more appropriate syntax error would be best. Imagine you write something like this:
```gleam
case list {
  [one, two, ..rest -> todo
}
```
Now this new error would pop up underlining the arrow and that would be quite confusing.
```
case list {
  [one, two, ..rest -> todo
                    ^^ 

I was expecting the end of the list.
The spread pattern precesing this consumes the remaining elements of the list
therefore there is nothing left to match.
```
The best approach here would be to copy what the parser is doing when parsing a list value: there we try and parse another list of items and see if it succeeds or not. So here we'd have to do something similar and parse a list of patterns and see if it succeeds (then we can show this error) or not (all good).